### DISCUSSION TITLE
Changing Image Colour

### DISCUSSION THREAD

**Discussion by lukaszossowski** [AvaloniaUI/Avalonia#11223] [Q&A] [answered]
Hey,

I would like to create a few controls that have icon images as content, but since their background will be changing I also need to change their colour from black to white and vice versa, depending on whether the button is pressed or not, like in this example:

![example](https://user-images.githubusercontent.com/112421929/236057501-3159dda2-2bbd-4116-8045-e11965541f8b.JPG)

How can I achieve that with Avalonia? I guess I could try doing some workarounds with 2 separate images, one white and second black, but that would be quite a lot of added complexity and inconvenience.

**Comment by timunie** [comment, marked as answer]
If the images are monochrome, I'd try to get the path data (from Inkscape for example) and use `Path` or `PathIcon` (v11 only). there you can switch the Foreground easily. 

Other options: 
- Use an Image lib that supports this
- Use tow images shown above each other and toggle their visibility

**Reply by lukaszossowski** [reply]
Thank you, I will take a look at PathIcon.

**Reply by timunie** [reply]
https://github.com/MahApps/IconPacks.Browser if you need this for getting the path data

**Reply by lukaszossowski** [reply]
Thanks, though with Inkscape and its bitmap tracing I was able to quickly convert some test icons and by saving them as xaml I could basically paste them straight to Avalonia.
